web/src/components/vis/ScorePlot.vue

```diff
@@ -91,23 +91,29 @@ export default {
       type: Boolean,
       default: true,
     },
-    /* {{
-     *   x: Array<Array<Number>>,
-     *   labels: Object<string, Array<string>>
-     * }}
-     */
-    rawPoints: {
+    pcCoords: {
       required: true,
-      validator: prop => !prop
-          || (typeof prop === 'object' && ['x', 'labels'].every(key => key in prop)),
+      type: Array,
+      validator: prop => prop.every(coord => coord.every(Number.isFinite)),
     },
-    dataset: {
+    rowLabels: {
       required: true,
-      validator: prop => typeof prop === 'object'
-        && '_source' in prop
-        && typeof prop._source === 'object'
-        && 'columns' in prop._source
-        && Array.isArray(prop._source.columns),
+      type: Array,
+      validator: prop => prop.every(val => typeof val === 'string'),
+    },
+    groupLabels: {
+      required: true,
+      type: Object,
+    },
+    eigenvalues: {
+      required: true,
+      type: Array,
+      validator: prop => prop.every(Number.isFinite),
+    },
+    columns: {
+      required: true,
+      type: Array,
+      validator: prop => prop.every(column => ['column_header', 'column_type'].every(key => Object.prototype.hasOwnProperty.call(column, key))),
```

Comment:
I've never given it much though before, but I'm cautions of O(N) validators.  I'm pretty sure validators don't run in production since they only produce warnings.

Reply:
This was just a musing, and can be ignored.

Reply:
Adopting typescript will moot this concern in any case.